### ISSUE TITLE
expose built-in commands as `core ...` in scripts to avoid shadowing

### ISSUE THREAD

**Issue by amtoine** [nushell/nushell#8854]
### Related problem

this is a discussion we had with @kubouch in an `std` PR about built-in command shadowing.

say you write a module called `foo` which has the `length` subcommand
1. you could define the command as `foo length`, use the module with `use foo *` and there would be no problem apart from the mandatory `*` in the `use` call
2. you could want to use `use foo` and have `foo length` available, not `foo foo length` => just rename `foo length` to `length` in the `foo` module right? but if `foo length` uses the builtin `length`, we have a problem...

### Describe the solution you'd like

an idea that came was the following
- have a module, a bit like `std`, called `core` with all the custom commands
- when you shadow a command with `def length [...] { ... }`, you can do `use core` and use `core length` for instance, which would call the builtin-length

**Comment by bobhy** [comment]
Great, so long as it is not necessary to `use core` to access `length` under normal circumstances.

**Comment by amtoine** [comment]
> Great, so long as it is not necessary to `use core` to access `length` under normal circumstances.
>
oooh no no, hell no :scream: 

is it unclear somewhere in the message? i do not want people to think that from what i've written :exploding_head:

**Comment by kubouch** [comment]
Yeah, we can split existing builtin Nushell commands into modules which would allow you to `use core; core length` if you run into the shadowing problem.
